client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { MEDIA_REQUEST } from 'state/action-types';
+import { receiveMedia, requestingMedia } from 'state/media/actions';
+import wpcom from 'lib/wp';
+
+/**
+ * Issues an API request to fetch media for a site and query.
+ *
+ * @param  {Object}         store  Redux store
+ * @param  {Object}         action Action object
+ * @return {XMLHttpRequest}        XMLHttpRequest
+ */
+export function requestMedia( { dispatch }, { siteId, query } ) {
+	dispatch( requestingMedia( siteId, query ) );
```

Comment:
Do you have on thoughts on whether the `isRequestingMedia` check currently in the query component would make more sense in the API middleware handler?